### ISSUE TITLE
"CRITICAL You shall not pass!" `np.zeros([len(indices)])`

### ISSUE THREAD

**Issue by paugier** [serge-sans-paille/pythran#728]
```
import numpy as np

# pythran export func(int32[])
def func(indices):
    arr = np.zeros([len(indices)])
    return arr
```

The compilation crashes... The error log is pretty long...

```
error: no match for call to ‘(pythonic::numpy::functor::zeros) (pythonic::types::list<long int>)’
...
template argument deduction/substitution failed
...
```

It looks like a bug...

Note that this works:

```
import numpy as np

# pythran export func(int32[])
def func(indices):
    arr = np.zeros((len(indices),))
    return arr
```

**Comment by serge-sans-paille** [comment]
That's a *known* bug. Basically it happens because we don't know statically the size of a list, so we cannot deduce the size of the array.

Yet we already have the concept of fixed size array, which could be useful here, I need some thinking on that.

**Comment by paugier** [comment]
Yes, even me, I remember now that I have already encountered this bug! Sorry to report a known bug. But still I managed to waste one hour to understand why pythran was unhappy with this module.

It is funny that we do not get this problem with
- `np.zeros((1000,))`
- `np.zeros(tuple([100]))`
- `np.zeros([100])`

but we get the bug for
- `np.zeros([1000])`

I figured out that it is because of pythran optimizations of the ast tree (i.e. of the python code).

- It is actually interesting to look at the optimized python code. There could be an option for that, similar to -E ? Serge, if you are ok with that, I can propose a PR because it is easy :-)

- Wouldn't it be good to have an optimization transforming the "constant" and "interned" lists in tuples?

- More generally, wouldn't be possible to detect pieces of code that will produce known bugs or errors and to print a nice explanation? At least for this case, this should not be technically difficult. It could be run only if we get an error. There is already such mechanism which gives 

```
E: Invalid argument type for function call to `Callable[[List[int]], ...]`, no overload found, tried:
...
```
If we could get the line of the error, it would be even better. Maybe even the name of the "callable" (here zeros)...
